### ISSUE TITLE
npm run functions:test fails

### ISSUE THREAD

**Issue by mradenovic** [prescottprue/generator-react-firebase#218]
**Describe the bug**
The default `npm run functions:test` script fails after install. Am I missing something in the setup?

`TypeError: Channel credentials must be a ChannelCredentials object` thrown every time firestore write is atempted.

There also dependencies that will probably cause fail of `deploy.yml` and `verify.yml` scripts.:
-  **nyc** in  ` "test:cov:base": "nyc --reporter=lcov npm run test:base"`
- **jq** in `"test:base": "cross-env FIRESTORE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.firestore.port') FIREBASE_DATABASE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.database.port') jest ./src/**/*.spec.js --config jest.config.js"` 

**To Reproduce**
1. Create function tests with `yo generator-firebase`
2. Run `npm run functions:test`
3. Run `yarn --cwd functions test:cov # Run unit tests with coverage`

**Expected behavior/code**
All tests run after install.

**Possible Solution**
Add `nyc` to dependencies. It looks like only`npm run functions:test` fails after installing it.

**Additional Context/Screenshots**
`npm run function:tests` output

```console
> mover-exchange@0.0.1 functions:test /home/mischa/dev/mover-exchange
> npm run test --prefix functions


> mover-exchange-functions@0.0.1 test /home/mischa/dev/mover-exchange/functions
> firebase emulators:exec --only firestore,database "npm run test:base"

i  emulators: Starting emulators: firestore, database
✔  hub: emulator hub started at http://localhost:4400
⚠  firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
i  firestore: firestore emulator logging to firestore-debug.log
✔  firestore: firestore emulator started at http://localhost:8080
i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
⚠  database: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
i  database: database emulator logging to database-debug.log
✔  database: database emulator started at http://localhost:9000
i  database: For testing set FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000
i  Running script: npm run test:base

> mover-exchange-functions@0.0.1 test:base /home/mischa/dev/mover-exchange/functions
> cross-env FIRESTORE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.firestore.port') FIREBASE_DATABASE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.database.port') jest ./src/**/*.spec.js --config jest.config.js

 FAIL  src/indexUser/indexUser.spec.js (5.019s)
  indexUser Firestore Cloud Function (onWrite)
    ✕ adds user to Firestore on create event (2401ms)
    ✕ updates existing user in Firestore on update event (14ms)
    ✕ removes user from Firestore on delete event (16ms)

  ● indexUser Firestore Cloud Function (onWrite) › adds user to Firestore on create event

    TypeError: Channel credentials must be a ChannelCredentials object

      at new ChannelImplementation (node_modules/google-gax/node_modules/@grpc/grpc-js/src/channel.ts:147:13)
      at new Client (node_modules/google-gax/node_modules/@grpc/grpc-js/src/client.ts:139:30)
      at new ServiceClientImpl (node_modules/google-gax/node_modules/@grpc/grpc-js/src/make-client.ts:116:3)
      at GrpcClient.createStub (node_modules/google-gax/src/grpc.ts:290:18)

  ● indexUser Firestore Cloud Function (onWrite) › updates existing user in Firestore on update event

    TypeError: Channel credentials must be a ChannelCredentials object

      at new ChannelImplementation (node_modules/google-gax/node_modules/@grpc/grpc-js/src/channel.ts:147:13)
      at new Client (node_modules/google-gax/node_modules/@grpc/grpc-js/src/client.ts:139:30)
      at new ServiceClientImpl (node_modules/google-gax/node_modules/@grpc/grpc-js/src/make-client.ts:116:3)
      at GrpcClient.createStub (node_modules/google-gax/src/grpc.ts:290:18)

  ● indexUser Firestore Cloud Function (onWrite) › removes user from Firestore on delete event

    TypeError: Channel credentials must be a ChannelCredentials object

      at new ChannelImplementation (node_modules/google-gax/node_modules/@grpc/grpc-js/src/channel.ts:147:13)
      at new Client (node_modules/google-gax/node_modules/@grpc/grpc-js/src/client.ts:139:30)
      at new ServiceClientImpl (node_modules/google-gax/node_modules/@grpc/grpc-js/src/make-client.ts:116:3)
      at GrpcClient.createStub (node_modules/google-gax/src/grpc.ts:290:18)

  console.error src/indexUser/index.js:63
    Error running updating displayName index for profile with userId: 123ABC Channel credentials must be a ChannelCredentials object

  console.log src/indexUser/index.js:24
    Profile being removed for user with id: 123ABC, removing from index...

  console.error src/indexUser/index.js:30
    Error running delete promises: Channel credentials must be a ChannelCredentials object

  console.log src/indexUser/index.js:24
    Profile being removed for user with id: 123ABC, removing from index...

  console.error src/indexUser/index.js:30
    Error running delete promises: Channel credentials must be a ChannelCredentials object

Test Suites: 1 failed, 1 total
Tests:       3 failed, 3 total
Snapshots:   0 total
Time:        5.068s
Ran all test suites matching /.\/src\/indexUser\/indexUser.spec.js/i.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mover-exchange-functions@0.0.1 test:base: `cross-env FIRESTORE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.firestore.port') FIREBASE_DATABASE_EMULATOR_HOST=localhost:$(cat ../firebase.json | jq '.emulators.database.port') jest ./src/**/*.spec.js --config jest.config.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mover-exchange-functions@0.0.1 test:base script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mischa/.npm/_logs/2020-05-05T16_20_35_101Z-debug.log
⚠  Script exited unsuccessfully (code 1)
i  emulators: Shutting down emulators.
i  hub: Stopping emulator hub
i  firestore: Stopping firestore emulator
i  database: Stopping database emulator

Error: Script "npm run test:base" exited with code 1
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mover-exchange-functions@0.0.1 test: `firebase emulators:exec --only firestore,database "npm run test:base"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mover-exchange-functions@0.0.1 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mischa/.npm/_logs/2020-05-05T16_20_36_633Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mover-exchange@0.0.1 functions:test: `npm run test --prefix functions`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mover-exchange@0.0.1 functions:test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mischa/.npm/_logs/2020-05-05T16_20_36_693Z-debug.log
```

**Comment by prescottprue** [comment]
Same as the other issue, we need to add the serviceAccount for admin access. I'll add a note about this to the README as well

**Comment by mradenovic** [comment]
Thanks a lot! I'm trying to learn about CI/CD and this project looks like a great place.

> The dependencies do not cause a failed verify or deploy since the image which github actions uses has jq installed...

How about `nyc`? The script doesn't got thru unless I install it manually.
Should `jq` go to README as a requirement? I work on debian container and it was not there by default :smile:

**Comment by prescottprue** [comment]
No - nyc is installed as an npm dependency through yarn though

**Comment by mradenovic** [comment]
I just realized that `nyc` is installed with `mocha` but not with `jest`. 

`test:cov:base": "nyc --reporter=lcov npm run test:base` is installed in both cases.

One of these is not right, not sure which one 😄

[package.json](https://github.com/prescottprue/generator-react-firebase/blob/master/generators/app/templates/functions/package.json)
```
# app/
    "firebase-functions-test": "^0.2.1",<% } %><% if(includeFunctionsTests && functionsTestTool === 'jest') { %>
    "jest": "^25.1.0",<% } %>
    "prettier": "^2.0.2",<% if(includeFunctionsTests && functionsTestTool === 'mocha') { %>
    "mocha": "^7.1.2",
    "nyc": "^15.0.0",<% } %>
```

**Comment by prescottprue** [comment]
Great catch - yeah it will have to be changed when using jest. Thought I had already added logic, but I'll look into that. No, it can stay in this issue since it all has to do with the functions tests

**Comment by mradenovic** [comment]
While we are at it, `test:ui:emulate` tries to run `test:open`instead of `test:ui:open`.

```diff
diff --git a/package.json b/package.json
index 1206d46..a57988a 100644
--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "functions:test": "npm run test --prefix functions",
     "test:ui": "cypress run",
     "test:ui:open": "cypress open",
-    "test:ui:emulate": "cross-env FIREBASE_DATABASE_EMULATOR_HOST=\"localhost:$(cat firebase.json | jq .emulators.database.port)\"  FIRESTORE_EMULATOR_HOST=\"localhost:$(cat firebase.json | jq .emulators.firestore.port)\" npm run test:open",
+    "test:ui:emulate": "cross-env FIREBASE_DATABASE_EMULATOR_HOST=\"localhost:$(cat firebase.json | jq .emulators.database.port)\"  FIRESTORE_EMULATOR_HOST=\"localhost:$(cat firebase.json | jq .emulators.firestore.port)\" npm run test:ui:open",
     "emulators": "firebase emulators:start --only firestore,database",
     "emulators:all": "firebase emulators:start --only firestore,database,hosting",
     "lint": "eslint .",
```

**Comment by prescottprue** [comment]
This was fixed v5.7.2, reach out if things aren't working as you expect after updating. Thanks for reporting with such detail!
